src/discord-api.ts

```diff
@@ -57,40 +65,53 @@ export const getNewData = async (env: Env) => {
 
 	const existingTopics: APIThreadChannel[] = []
 
-	//Get the topics with new messages
-	const newMessagesTopics = allTopics
-		.map((thread) => {
-			const lastSavedMessage = savedTopics?.find(
-				(t) => t.id === thread.id
-			)?.last_message_id
-
-			if (thread.last_message_id !== lastSavedMessage) {
-				// We need to update the last_message_id for this topic
-				existingTopics.push(thread)
-				return { id: thread.id, after: lastSavedMessage }
-			}
-		})
-		.filter((m) => m !== undefined)
+	const newMessages: APIMessage[] = []
+	const updatedMessages: APIMessage[] = []
+
+	const delay = (ms: number) => new Promise((res) => setTimeout(res, ms))
 
 	// Get messages on a topic
 	// We can't make more then 50 requests per second to Discord API. So we will wait 1 second after each 40 requests
-	const newMessages: APIMessage[] = []
-	const delay = (ms: number) => new Promise((res) => setTimeout(res, ms))
+	for (let i = 0; i < allTopics.length; i += 40) {
+		const threads = allTopics.slice(i, i + 40)
+		const threadsIds = threads.map((t) => t.id)
+
+		const savedMessages =
+			(
+				await supabase
+					.from('messages')
+					.select('id,updated_at')
+					.in('topic_id', threadsIds)
+			).data ?? []
+
+		const messages = await getMessages(threads, env)
+		const nMessages = messages.filter(
+			(m) => !savedMessages.find((sm) => sm.id === m.id)
+		)
+		const uMessages = messages.filter((m) => {
+			const savedMsg = savedMessages.find((sm) => sm.id === m.id)
+
+			if (savedMsg?.updated_at && m.edited_timestamp) {
+				if (new Date(m.edited_timestamp) > new Date(savedMsg.updated_at)) {
```

Comment:
I guess this would also mean you should update?
```suggestion
				if (new Date(m.edited_timestamp) !== new Date(savedMsg.updated_at)) {
```